pure_pursuit_wrapper/src/pure_pursuit_wrapper.cpp

```diff
@@ -73,15 +73,14 @@ void PurePursuitWrapper::TrajectoryPlanPoseHandler(const geometry_msgs::PoseStam
 
 };
 
-autoware_msgs::Waypoint PurePursuitWrapper::TrajectoryPlanPointToWaypointConverter(geometry_msgs::PoseStamped pose, cav_msgs::TrajectoryPlanPoint tpp) {
+autoware_msgs::Waypoint PurePursuitWrapper::TrajectoryPlanPointToWaypointConverter(double current_time, geometry_msgs::PoseStamped pose, cav_msgs::TrajectoryPlanPoint tpp) {
   ROS_DEBUG_STREAM("Convertering TrajectoryPlanPointToWaypoint");
 
   autoware_msgs::Waypoint waypoint;
 
   waypoint.pose.pose.position.x = tpp.x;
   waypoint.pose.pose.position.y = tpp.y;
-  double begin = ros::Time::now().toSec();
-  double delta_t = tpp.target_time - (begin * 1000);
+  double delta_t = (tpp.target_time / 1000) - (current_time);
```

Comment:
if target_time is a double please divide by 1000.0. This makes it clearer and ensures there will be no truncation.